src/Swift.Bindings/src/TypeDatabase/ITypeDatabase.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace BindingsGeneration;
+
+/// <summary>
+/// Type database. Holds module databases and type records for Swift types.
+/// </summary>
+public interface ITypeDatabase
+{
+    /// <summary>
+    /// Loads a module database from a specified file.
+    /// </summary>
+    /// <param name="file">The file path of the module database to load.</param>
+    public void LoadModuleDatabaseFromFile(string file);
```

Comment:
Since I/O is slow, let's make this async.
Do we need a stream variant?

Reply:
Made it async, parsing xml is still synchronous so there is only partial gain. I think we can revisit this one in the future.